### PR TITLE
Update region filters in Bangladesh December experiments

### PR DESCRIPTION
**Pull request by qptr**
**Story card:** [sc-11379](https://app.shortcut.com/simpledotorg/story/11379)

This PR adds a data migration file to update region filters of the December experiments set up in Bangladesh. Spec added to test the up and down migrations.

### CODE REVIEW

File: db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    facilities = {
+      sylhet: %w[uhc-osmaninagar uhc-taherpur],
+      rajshahi: %w[atgharia-upazila-health-complex uhc-bera uhc-faridpur uhc-sujanagar-4b973cb1-d584-4100-91ca-778f218b502],
+      netrokona: %w[uhc-barhatta uhc-durgapur-dbd6efea-8950-410e-abc3-ed9a329fb9d3 uhc-kalmakanda uhc-kendua uhc-khaliajuri uhc-madan uhc-purbadhala],
+      sherpur: %w[uhc-jhenaigati uhc-nakhla uhc-nalitabari uhc-sribordi],
+      jamalpur: %w[uhc-bakshiganj uhc-dewanganj uhc-islampur uhc-madarganj uhc-melandah],
+      feni: %w[uhc-chhagalniya uhc-daganbhuiya uhc-fulgazi uhc-parsuram uhc-sonagazi],
+      chattogram: %w[uhc-anwara uhc-banshkhali uhc-boalkhali uhc-chandanaish uhc-fatikchori uhc-hathazari uhc-karnaphuli uhc-lohagara uhc-mirsharai uhc-patiya uhc-rangunia uhc-raozan uhc-sandwip uhc-satkania uhc-sitakunda],
+      bandarban: ["uhc-lama"],
+      barishal: %w[uhc-kathalia-86845c14-b23d-4578-820a-63eb9628422a uhc-nalchity-97cb9397-8ca7-4edb-81ae-2ac8fb1ea93a uhc-rajapur-40d0e15a-37c4-4833-8f55-446bb210c457 uhc-agailjhara uhc-bakerganj uhc-gaurnadi uhc-hijla uhc-mehendiganj uhc-muladi uhc-wazirpur uhc-babuganj]
```
Comment (tfidfwastaken) [inline db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb:16]: The [story card](https://app.shortcut.com/simpledotorg/story/11379/setting-up-new-sms-experiments-in-bangladesh) has 13 districts listed, but here I see only 9 districts, and they don't exactly line up (for eg, this one has Sylhet, but the list in the story card does not). Am I missing something here?

File: db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                              uhc-goshairhat uhc-sreenagar uhc-tongibari uhc-sirajdikhan uhc-hathazari uhc-fatikchori uhc-anwara uhc-satkania
+                              uhc-lohagara uhc-patiya uhc-rangunia uhc-sitakunda uhc-chandanaish uhc-raozan uhc-banshkhali uhc-boalkhali uhc-karnaphuli
+                              uhc-mirsharai uhc-sandwip uhc-fulgazi uhc-parsuram uhc-daganbhuiya uhc-sonagazi uhc-chhagalniya uhc-atpara uhc-barhatta
+                              uhc-kalmakanda uhc-kendua uhc-khaliajuri uhc-madan uhc-mohanganj uhc-purbadhala uhc-durgapur-dbd6efea-8950-410e-abc3-ed9a329fb9d3
+                              uhc-belkuchi uhc-chowhali uhc-kamarkhanda uhc-kazipur uhc-shahzadpur uhc-tarash uhc-ullapara uhc-bera uhc-bhangura
+                              uhc-faridpur uhc-iswardi uhc-santhia uhc-chatmohar uhc-sujanagar-4b973cb1-d584-4100-91ca-778f218b502e atgharia-upazila-health-complex
+                              uhc-islampur uhc-melandah uhc-sarishabari uhc-dewanganj uhc-bakshiganj uhc-madarganj uhc-jhenaigati uhc-nalitabari uhc-nakhla
+                              uhc-sribordi]}}
```
Comment (tfidfwastaken) [inline db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb:40]: nit: indentation

```suggestion
    old_region_filters = {
      "facilities" =>
        {"include" => %w[uhc-agailjhara uhc-babuganj-facility uhc-bakerganj uhc-hijla uhc-mehendiganj
          uhc-muladi uhc-wazirpur uhc-kathalia-86845c14-b23d-4578-820a-63eb9628422a uhc-nalchity-97cb9397-8ca7-4edb-81ae-2ac8fb1ea93a
          uhc-rajapur-40d0e15a-37c4-4833-8f55-446bb210c457 thc-tejgaon-sadar uhc-dhamrai uhc-dohar uhc-nawabganj uhc-damuddya
          uhc-goshairhat uhc-sreenagar uhc-tongibari uhc-sirajdikhan uhc-hathazari uhc-fatikchori uhc-anwara uhc-satkania
          uhc-lohagara uhc-patiya uhc-rangunia uhc-sitakunda uhc-chandanaish uhc-raozan uhc-banshkhali uhc-boalkhali uhc-karnaphuli
          uhc-mirsharai uhc-sandwip uhc-fulgazi uhc-parsuram uhc-daganbhuiya uhc-sonagazi uhc-chhagalniya uhc-atpara uhc-barhatta
          uhc-kalmakanda uhc-kendua uhc-khaliajuri uhc-madan uhc-mohanganj uhc-purbadhala uhc-durgapur-dbd6efea-8950-410e-abc3-ed9a329fb9d3
          uhc-belkuchi uhc-chowhali uhc-kamarkhanda uhc-kazipur uhc-shahzadpur uhc-tarash uhc-ullapara uhc-bera uhc-bhangura
          uhc-faridpur uhc-iswardi uhc-santhia uhc-chatmohar uhc-sujanagar-4b973cb1-d584-4100-91ca-778f218b502e atgharia-upazila-health-complex
          uhc-islampur uhc-melandah uhc-sarishabari uhc-dewanganj uhc-bakshiganj uhc-madarganj uhc-jhenaigati uhc-nalitabari uhc-nakhla
          uhc-sribordi]}
    }

```

**Review by tfidfwastaken — COMMENTED** [review]
There appears to be a mismatch in the facility list between the story card and in this PR. Other than that this data migration itself LGTM. Nice spec too.

File: db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb (hunk 1 above)
Comment (qptr) [reply db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb:16]: There was an update to the facility list based on medicine availability. I've updated the card.

Comment (qptr) [reply db/data/20231127065753_update_facility_list_for_bd_december_experiment.rb:16]: Auditing the facility list in the migration file now

**Review by tfidfwastaken — APPROVED** [review]

**Pull request merged**